src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3069 @@
+    data->narcA180 = NARC_New(NARC_a_1_8_0, HEAP_ID_REGISTER_HALL_OF_FAME);
+    RegisterHallOfFame_GetPartyDetails(data);
+    RegisterHallOfFame_SetGfxBanks();
+    RegisterHallOfFame_CreateBgConfig(data);
+    RegisterHallOfFame_CreateSpriteGfxHandlers(data);
+    sub_02004EC4(8, SEQ_GS_E_DENDOUIRI, 1);
+    sub_02004EC4(71, 0, 0);
+    LoadFontPal0(GF_PAL_LOCATION_MAIN_BG, (enum GFPalSlotOffset)0x1E0, HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->currentScene = REGHOF_SCENE_INDIV_MONS_INIT;
+    return TRUE;
+}
+
+BOOL RegisterHallOfFame_Exit(OVY_MANAGER *man, int *state) {
+    RegisterHallOfFameData *data = OverlayManager_GetData(man);
+    RegisterHallOfFame_DestroySpriteGfxHandlers(data);
+    RegisterHallOfFame_DestroyBgConfig(data);
+    NARC_Delete(data->narcA180);
+    NARC_Delete(data->narcA094);
+    String_Delete(data->strbuf1);
+    String_Delete(data->strbuf2);
+    MessageFormat_Delete(data->msgFormat);
+    DestroyMsgData(data->msgData);
+    OverlayManager_FreeData(man);
+    DestroyHeap(HEAP_ID_REGISTER_HALL_OF_FAME);
+    return TRUE;
+}
+
+static RegisterHallOfFameScene (*const sSceneFuncs[8])(RegisterHallOfFameData *data);
+
+BOOL RegisterHallOfFame_Main(OVY_MANAGER *man, int *state) {
+    RegisterHallOfFameData *data = OverlayManager_GetData(man);
+    data->currentScene = sSceneFuncs[data->currentScene](data);
+    if (data->currentScene == REGHOF_SCENE_MAX) {
+        return TRUE;
+    }
+    RegisterHallOfFame_PicAnimationsTick(data);
+    return FALSE;
+}
+
+static void RegisterHallOfFame_SetGfxBanks(void) {
+    GfGfx_SetBanks(&sGfxBanks);
+}
+
+static void VBlankTask_RegisterHallOfFame_IndividualMonsCongrats(SysTask *task, void *taskData) {
+    RegisterHallOfFameData *data = (RegisterHallOfFameData *)taskData;
+    if (data->subprocCallback != NULL && !data->subprocCallback(data)) {
+        data->subprocCallback = NULL;
+    }
+    DoScheduledBgGpuUpdates(data->bgConfig);
+    sub_0200D020(data->spriteGfxHandler);
+    thunk_OamManager_ApplyAndResetBuffers();
+    OS_SetIrqCheckFlag(OS_IE_V_BLANK);
+}
+
+static void RegisterHallOfFame_ReplaceSpriteChar(const void *pSrc, u32 offset, u32 size) {
+    DC_FlushRange(pSrc, size);
+    GX_LoadOBJ(pSrc, offset, size);
+}
+
+static void RegisterHallOfFame_ReplaceSpritePltt(const void *pSrc, u32 offset, u32 size) {
+    DC_FlushRange(pSrc, size);
+    GX_LoadOBJPltt(pSrc, offset, size);
+}
+
+static void RegisterHallOfFame_CreateBgConfig(RegisterHallOfFameData *data) {
+    data->bgConfig = BgConfig_Alloc(HEAP_ID_REGISTER_HALL_OF_FAME);
+    BG_SetMaskColor(GF_BG_LYR_SUB_0, RGB_BLACK);
+}
+
+static void RegisterHallOfFame_DestroyBgConfig(RegisterHallOfFameData *data) {
+    FreeToHeap(data->bgConfig);
+}
+
+static void RegisterHallOfFame_CreateSpriteGfxHandlers(RegisterHallOfFameData *data) {
+    data->spriteRenderer = SpriteRenderer_Create(HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->spriteGfxHandler = SpriteRenderer_CreateGfxHandler(data->spriteRenderer);
+
+    {
```

Comment:
unnecessary(?) scope brackets

Reply:
the original intent was to inline this data, but doing so while maintaining link order prooved too frustrating. nevertheless, while the scope braces are not necessary to match, i do find them necessary for semantics.